sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -138,9 +138,12 @@ class SQLBuilder private (
     case g: Generate =>
       generateToSQL(g)
 
-    case Limit(limitExpr, child) =>
+    case Limit(limitExpr, child: SubqueryAlias) =>
```

Comment:
This makes `((~) AS gen_subquery_0 LIMIT 2)` pattern.
For example, `select * from ((select 1) as T)` works, but `select * from ((select 1) as T limit 2)` fails.